transport/transport.go

```diff
@@ -506,3 +507,50 @@ func wait(ctx context.Context, closing <-chan struct{}, proceed <-chan int) (int
 		return i, nil
 	}
 }
+
+const (
+	spaceByte   = byte(int(' '))
+	tildaByte   = byte(int('~'))
+	percentByte = byte(int('%'))
+)
+
+// matching https://github.com/grpc/grpc-java/pull/1517/files
+func grpcMessageEncode(grpcMessage string) string {
+	if grpcMessage == "" {
+		return ""
+	}
+	var buffer bytes.Buffer
+	for _, c := range []byte(grpcMessage) {
+		if c >= spaceByte && c < tildaByte && c != percentByte {
+			_ = buffer.WriteByte(c)
```

Comment:
It's for `errcheck`, and a style that I like - by doing a blank assignment, I'm explicitly saying "I know there is an error returned here, and I'm purposefully ignoring it, this is not a bug". It tells future contributors that this is not an issue that should be checked, that we are extremely confident this error will not happen and that checking it would be not worthwhile and get in the way of cleaner code (ie no returned error from this function). Does that make sense?